src/java.desktop/share/native/libjsound/MidiOutDevice.c

```diff
@@ -94,7 +94,7 @@ Java_com_sun_media_sound_MidiOutDevice_nGetTimeStamp(JNIEnv* e, jobject thisObj,
 
     /* Handle error codes. */
     if (ret < -1) {
-        ERROR1("Java_com_sun_media_sound_MidiOutDevice_nGetTimeStamp: MIDI_IN_GetTimeStamp returned %lld\n", ret);
+        ERROR1("Java_com_sun_media_sound_MidiOutDevice_nGetTimeStamp: MIDI_IN_GetTimeStamp returned %lld\n", (long long int) ret);
```

Comment:
I wonder if we could use the format specifier for jlong here somehow, avoiding the cast here

Reply:
I think we do not have this now in this lib. Of course we can copy something like this

test/hotspot/jtreg/vmTestbase/nsk/share/native/nsk_tools.hpp 
```
30  #include <inttypes.h>

34  #if defined(_LP64) && defined(__APPLE__)
35  #define JLONG_FORMAT "%ld"
36  #else  // _LP64 && __APPLE__
37  #define JLONG_FORMAT "%" PRId64
38  #endif // _LP64 && __APPLE__
```

into the file and then use JLONG_FORMAT .
Is this worth it?

Reply:
That seems preferable, but I'll leave it up to you and the other people who reviewed this on where to put the define, whether locally in the file, or in a header